eZ/Publish/Core/MVC/Symfony/View/Builder/ContentViewBuilder.php

```diff
@@ -100,6 +100,12 @@ public function buildView(array $parameters)
         } elseif ($location instanceof Location) {
             // if we already have location load content true it so we avoid dual loading in case user does that in view
             $content = $location->getContent();
+            if (!$this->canRead($content, $location)) {
+                throw new UnauthorizedException(
```

Comment:
@crevillo no, the "main" content will be displayed normally. In the place where embedded content should be displayed, an information about the lack of permissions will be shown. So, there is no change of behavior in case of embeds. 

![screen shot 2018-08-16 at 10 03 33](https://user-images.githubusercontent.com/10659510/44196251-acdb5d80-a13b-11e8-9693-d8c0da4b9b7e.png)

> As for loadContent() (which will throw from API) I assume some exception conversion from API exceptions will happens somewhere else, right @kmadejski ?

Yes, `UnauthorizedException` thrown in API are handled by https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Bundle/EzPublishCoreBundle/EventListener/ExceptionListener.php.
The issue here is, that the `UnauthorizedException` is thrown too late, when Twig tries to display template and therefore the exception isn't handled properly.